beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1669,6 +1721,164 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         );
         Ok(())
     }
+
+    ///  Try to prune blobs, approximating the current epoch from lower epoch numbers end (older
+    /// end) and is useful when the data availability boundary is not at hand.
+    pub fn try_prune_most_blobs(&self, force: bool) -> Result<(), Error> {
+        let eip4844_fork = match self.spec.eip4844_fork_epoch {
+            Some(epoch) => epoch,
+            None => {
+                debug!(self.log, "Eip4844 fork is disabled");
+                return Ok(());
+            }
+        };
+        // At best, current_epoch = split_epoch + 2. However, if finalization doesn't advance, the
+        // `split.slot` is not updated and current_epoch > split_epoch + 2.
+        let min_current_epoch = self.get_split_slot().epoch(E::slots_per_epoch()) + Epoch::new(2);
+        let min_data_availability_boundary = std::cmp::max(
+            eip4844_fork,
+            min_current_epoch.saturating_sub(*MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS),
+        );
+
+        self.try_prune_blobs(force, Some(min_data_availability_boundary))
+    }
+
+    /// Try to prune blobs older than the data availability boundary.
+    pub fn try_prune_blobs(
+        &self,
+        force: bool,
+        data_availability_boundary: Option<Epoch>,
+    ) -> Result<(), Error> {
+        let (data_availability_boundary, eip4844_fork) =
+            match (data_availability_boundary, self.spec.eip4844_fork_epoch) {
+                (Some(boundary_epoch), Some(fork_epoch)) => (boundary_epoch, fork_epoch),
+                _ => {
+                    debug!(self.log, "Eip4844 fork is disabled");
+                    return Ok(());
+                }
+            };
+
+        let should_prune_blobs = self.get_config().prune_blobs;
+        if !should_prune_blobs && !force {
+            debug!(self.log, "Blob pruning is disabled";
+                "prune_blobs" => should_prune_blobs
+            );
```

Comment:
Lets wrap this more like

```suggestion
            debug!(
                self.log,
                "Blob pruning is disabled";
                "prune_blobs" => should_prune_blobs
            );
```